retrofit2/src/test/java/com/linecorp/armeria/client/retrofit2/RetrofitClassAwareMeterIdPrefixFunctionTest.java

```diff
@@ -0,0 +1,241 @@
+import com.linecorp.armeria.client.metric.MetricCollectingClient;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.metric.MeterIdPrefixFunction;
+import com.linecorp.armeria.common.metric.MoreMeters;
+import com.linecorp.armeria.common.metric.NoopMeterRegistry;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.testing.junit.server.ServerExtension;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import retrofit2.http.DELETE;
+import retrofit2.http.GET;
+import retrofit2.http.HEAD;
+import retrofit2.http.HTTP;
+import retrofit2.http.OPTIONS;
+import retrofit2.http.PATCH;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+
+public class RetrofitClassAwareMeterIdPrefixFunctionTest {
+
+    private static final SimpleMeterRegistry meterRegistry = new SimpleMeterRegistry();
+    private static final ClientFactory clientFactory = ClientFactory.builder()
+                                                                    .meterRegistry(meterRegistry)
+                                                                    .build();
+
+    interface Example {
+        @DELETE("/foo")
+        CompletableFuture<Void> deleteFoo();
+
+        @GET("/foo")
+        CompletableFuture<Void> getFoo();
+
+        @HEAD("/foo")
+        CompletableFuture<Void> headFoo();
+
+        @HTTP(method = "TRACE", path = "/foo")
+        CompletableFuture<Void> traceFoo();
+
+        @OPTIONS("/foo")
+        CompletableFuture<Void> optionsFoo();
+
+        @PATCH("/foo")
+        CompletableFuture<Void> patchFoo();
+
+        @POST("/foo")
+        CompletableFuture<Void> postFoo();
+
+        @PUT("/foo")
+        CompletableFuture<Void> putFoo();
+    }
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.service("/foo", new AbstractHttpService() {
+                @Override
+                protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+
+                @Override
+                protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+
+                @Override
+                protected HttpResponse doTrace(ServiceRequestContext ctx, HttpRequest req) {
+                    return HttpResponse.of(HttpStatus.OK);
+                }
+            });
```

Comment:
nit: 
1) If you want to handle all methods
```java
sb.service("/foo", (ctx, req) -> HttpResponse.of(HttpStatus.OK));
```
2) If you want to handle `GET`, `POST` and `TRACE` together
```java
sb.service(Route.builder().path("/foo")
                .methods(HttpMethod.GET, HttpMethod.POST, HttpMethod.TRACE)
                .build(),
           (ctx, req) -> HttpResponse.of(HttpStatus.OK));
```